addons/website/models/website.py

```diff
@@ -190,9 +191,11 @@ def _handle_favicon(self, vals):
             vals['favicon'] = tools.image_process(vals['favicon'], size=(256, 256), crop='center', output_format='ICO')
 
     def unlink(self):
-        website = self.search([('id', 'not in', self.ids)], limit=1)
-        if not website:
-            raise UserError(_('You must keep at least one website.'))
+        if not self.env.context.get(MODULE_UNINSTALL_FLAG, False):
```

Comment:
Would have been the same but nitpicking and already r+ :yum: 
```suggestion
        if not self.env.context.get(MODULE_UNINSTALL_FLAG):
```

Reply:
I prefer with the "False" anyway, I would not have put it otherwise ;)

> Either the last one being deleted is a manually created one, and in this case it simply doesn't go through the unlink() method but is deleted through the table deletion, see delete(self.env['ir.model'].browse(model_ids))

What do you mean? If it does not go through unlink, then my fix is not good enough?

Reply:
Unlink is not call for a module's record when you remove that module, except for its records linked to an xml_id.
It is just done in SQL when the table is dropped, not through the ORM.

Reply:
Indeed, this should be done in the uninstall hook to cover all the cases actually :grimacing: